### PR TITLE
Extends view to support view names, component names and selectors...

### LINKED ISSUE #4: Questions: Production builds & supporting component(name)
Thanks for this addon, looks great. I have a few questions if you don't mind.

* How is this currently included in a project? If I add it to my project will this appear in the production `vendor.js` assets? If it does, can it be configured to only appear in development and testing?
* I notice you don't currently support `component(name)`, was there a reason for this and if not, would you be happy for me to submit a PR to support it?

### PR DESCRIPTION
**Pull request by MiguelMadero**
...  Adds a views method to support multiple matches when using a selector or view name.

Fixes #4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,11 +57,22 @@ Returns the model for the named controller. `name` defaults to the the current r
...
+
+### `views: function(viewNameOrSelector)`
+
+Returns all view instances for a given type or selector `select` or `.ember-select`. If the string resolves to a
+component or view name then returns all the views of that type or its sub-types. Otherwise the string is used as a selector.
+If nothing matches an empty array is returned.
+
+### `components: function(viewNameOrSelector)`
+
+Alias for `views`, since `Ember.Component` inherits from `Ember.View`.
```
Comment (aexmachina) [inline README.md]: It looks to me that this isn't defined.

File: app/services/ember-devtools.js [hunk 2]
```diff
@@ -40,11 +43,28 @@ export default Ember.Object.extend({
...
-      id = id.id;
+  view: function(idOrSelector) {
+    if (typeof idOrSelector === 'object') {
+      idOrSelector = idOrSelector.id;
+    }
+    return Ember.View.views[idOrSelector] || this.views(idOrSelector)[0];
+  },
+  views: function (selectorOrName) {
+    var views = Ember.View.views,
+      viewClass =  this.lookupFactory('component:' + selectorOrName) || this.lookupFactory('view:' + selectorOrName);
```
Comment (aexmachina) [inline app/services/ember-devtools.js]: Separate `var` on new lines please :)

File: app/services/ember-devtools.js [hunk 3]
```diff
@@ -40,11 +43,28 @@ export default Ember.Object.extend({
...
+      }).filter(function (view) {
+        return view instanceof viewClass;
+      });
     }
-    return Ember.View.views[id];
+
+    // We found elements via the selector, use that. Otherwise, try witha  name.
+    return map.call($(selectorOrName), function (element) {
+      return views[element.id];
+    }, this);
```
Comment (aexmachina) [inline app/services/ember-devtools.js]: `this` is redundant?

File: app/services/ember-devtools.js [hunk 4]
```diff
@@ -40,11 +43,28 @@ export default Ember.Object.extend({
...
+    if (viewClass) {
+      return Object.keys(views).map(function (id) {
+        return views[id];
+      }).filter(function (view) {
+        return view instanceof viewClass;
+      });
     }
-    return Ember.View.views[id];
+
+    // We found elements via the selector, use that. Otherwise, try witha  name.
```
Comment (aexmachina) [inline app/services/ember-devtools.js]: I think this comment is not correct?

**Comment by aexmachina** [comment]
Nice one! See feedback in comments, but otherwise good to :airplane:.

File: README.md (hunk 1 above)
Comment (MiguelMadero) [inline README.md]: fixed.

File: app/services/ember-devtools.js (hunk 2 above)
Comment (MiguelMadero) [inline app/services/ember-devtools.js]: It's a single var, just an or. I changed it to use multiple vars for consistency with line 117

File: app/services/ember-devtools.js (hunk 3 above)
Comment (MiguelMadero) [inline app/services/ember-devtools.js]: good catch. Thanks

**Comment by MiguelMadero** [comment]
@aexmachina thanks for the feedback. Sorry for the reply. I lost this in my inbox. I did some changes, rebased and pushed.

**Comment by aexmachina** [comment]
Nice one.

**Comment by aexmachina** [comment]
Published v3.1.0

**Comment by MiguelMadero** [comment]
Thanks



—

**Pull request merged**